### ISSUE TITLE
Zone activities farm plots issue

### ISSUE THREAD

**Issue by adamkad1** [CleverRaven/Cataclysm-DDA#47737]
# Describe the bug 
After finishing with farming, if there are sort zones nearby, the character will being tidying up, but sometimes, they go to places with no zones in them and do... something? idk. 

# Steps To Reproduce

Make farm plot zone
make sort zone
commence autofarm
watch your character maybe do something they shouldnt do

# Expected behavior
Not going to places without zones and doing stuff

# Screenshots

# Versions and configuration

Tried on dda only, build 11495

# Additional context
I wish you could just turn off 'tidy after autofarm'

**Comment by Kelenius** [comment]
Yeah, I had a similar issue. Can't provide screenshots or save file ATM, but my character started moving wood logs from outside of any zones into a food zone for some reason after finishing working on the farm. They were pretty far from the farms.

**Comment by PatrikLundell** [comment]
I've seen some odd "tidying" action after e.g. vehicle dismantling in 0.E2 stable where the companions sometimes move things into weird zones (like the unsorted goods zones in the "real" vehicle, or into the walls of the central building (there's a camp usage zone covering everything, including the walls, but there are no "normal" storage zones there).

**Comment by zeropol** [comment]
I've encountered the same issue not long ago ( november 2020), after farming, the character sort random loot that isnt in any zone.
I've not tested with the current version.
